File: frontend/src/types/graph.ts
export interface GraphNode {
  id: string;
  type: string;
  label: string;
  metadata: Record<string, unknown>;
}

export interface GraphEdge {
  id: string;
  source: string;
  target: string;
  relation: string;
  metadata: Record<string, unknown>;
}

export interface EvidenceGraph {
  scope: string;
  nodes: GraphNode[];
  edges: GraphEdge[];
}

File: backend/app/schemas/graph.py
from typing import Any

from pydantic import BaseModel, Field


class GraphNode(BaseModel):
    id: str
    type: str
    label: str
    metadata: dict[str, Any] = Field(default_factory=dict)


class GraphEdge(BaseModel):
    id: str
    source: str
    target: str
    relation: str
    metadata: dict[str, Any] = Field(default_factory=dict)


class EvidenceGraphResponse(BaseModel):
    scope: str
    nodes: list[GraphNode]
    edges: list[GraphEdge]
